platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/events/performance/MemoryMetricsProvider.kt

```diff
@@ -9,61 +9,74 @@ package io.bitdrift.capture.events.performance
 
 import android.app.ActivityManager
 import android.os.Debug
-import io.bitdrift.capture.common.RuntimeFeature
+import io.bitdrift.capture.common.RuntimeConfig
 
 private const val KB = 1024L
 
 internal class MemoryMetricsProvider(
     private val activityManager: ActivityManager,
+    private val jvmMemoryProvider: JvmMemoryProvider = DefaultJvmMemoryProvider(),
 ) : IMemoryMetricsProvider {
     var runtime: io.bitdrift.capture.common.Runtime? = null
 
-    // We only save the threshold on first access since it's a constant value obtained via a rather expensive Binder call
-    private val memoryThresholdBytes: Long by lazy {
-        return@lazy if (runtime?.isEnabled(RuntimeFeature.APP_MEMORY_PRESSURE) != false) {
-            ActivityManager
-                .MemoryInfo()
-                .also { memoryInfo ->
-                    activityManager.getMemoryInfo(memoryInfo)
-                }.threshold
-        } else {
-            // Use sentinel value
-            Long.MAX_VALUE
-        }
+    private val appLowMemoryConfigThreshold by lazy {
+        getConfiguredLowMemoryPercentThreshold()
     }
 
     override fun getMemoryAttributes(): Map<String, String> =
         buildMap {
-            put("_jvm_used_kb", usedJvmMemoryBytes().bToKb())
-            put("_jvm_total_kb", totalJvmMemoryBytes().bToKb())
-            put("_jvm_max_kb", maxJvmMemoryBytes().bToKb())
+            put("_jvm_used_kb", jvmMemoryProvider.usedMemoryBytes().bToKb())
+            put("_jvm_total_kb", jvmMemoryProvider.totalMemoryBytes().bToKb())
+            put("_jvm_max_kb", jvmMemoryProvider.maxMemoryBytes().bToKb())
             put("_native_used_kb", allocatedNativeHeapSizeBytes().bToKb())
             put("_native_total_kb", totalNativeHeapSizeBytes().bToKb())
             put("_memory_class", memoryClassMB().toString())
-            memoryThresholdBytes.takeIf { it != Long.MAX_VALUE }?.let {
-                put("_threshold_mem_kb", memoryThresholdBytes.bToKb())
-                put("_is_memory_low", if (isMemoryLow()) "1" else "0")
-            }
+            put("_is_memory_low", if (isMemoryLow()) "1" else "0")
+            put("_jvm_used_percent", "%.3f".format(jvmUsedPercent()))
         }
 
     override fun getMemoryClass(): Map<String, String> = buildMap { put("_memory_class", memoryClassMB().toString()) }
 
     override fun isMemoryLow(): Boolean {
-        val totalUsedMemoryByApp = usedJvmMemoryBytes() + allocatedNativeHeapSizeBytes()
-        return totalUsedMemoryByApp > memoryThresholdBytes
+        val thresholdPercent = appLowMemoryConfigThreshold ?: return false
```

Comment:
we should probably not report the field al together if it's disabled (that way those data points won't be considered at all in the charts)